java/src/main/native/src/RmmJni.cpp

```diff
@@ -344,6 +343,7 @@ JNIEXPORT void JNICALL Java_ai_rapids_cudf_Rmm_initializeInternal(JNIEnv *env, j
 
     bool use_pool_alloc = allocation_mode & 1;
     bool use_managed_mem = allocation_mode & 2;
+    bool use_arena_alloc = allocation_mode & 3;
```

Comment:
Similar to my comment above, with `ARENA = 0x00000003` that means `use_pool_alloc` will always be true when trying to specify the arena allocator and it will always take the pool allocator path instead of the arena allocator path.

Note that once the value is corrected for `ARENA` then the Java code should sanity check that the caller isn't trying to request `POOL` and `ARENA` at the same time.

Reply:
Fixed.